docs/style/CODING_STYLE_GUIDE.md

```diff
@@ -0,0 +1,167 @@
+| C++      | C++17   |
+| Python   | 3.10    |
+
+Product-specific software may elect to use later standards in their own work.
+
+## Coding Guidelines
+
+### Common
+
+#### When in Rome
+
+The most important convention and practice in the Matter SDK repo is "_When in
+Rome..._", per the quote below.
+
+[quote, St. Ambrose]
+
+---
+
+If you should be in Rome, live in the Roman manner; if you should be elsewhere,
+live as they do there.
+
+---
+
+Your extensions or fixes to existing code should match the prevailing of the
+original code.
+
+If you find the conventions so foreign or otherwise confusing, it may be best to
+let whoever owns the file make the necessary changes or seek the counsel of
+others in the group to find out what the right thing to do is. Never just start
+changing code wholesale for personal reasons without consulting others first.
+
+#### Commenting Out or Disabling Code
+
+Unused code shall not be disabled by commenting it out with C- or C++-style
+comments or with preprocessor `#if 0 ... #endif` semantics. Unused code should
+be removed.
+
+#### Auto-formatters
+
+We use the following auto-formatters on code:
+
+| Language   | Formatter          | Style File                                                                                                                                                                                                                                                                |
+| ---------- | ------------------ | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| C++        | clang-format       | [.clang-format](https://github.com/project-chip/connectedhomeip/blob/master/.clang-format)                                                                                                                                                                                |
+| ObjectiveC | clang-format       | [.clang-format](https://github.com/project-chip/connectedhomeip/blob/master/.clang-format)                                                                                                                                                                                |
+| java       | google-java-format | N/A                                                                                                                                                                                                                                                                       |
+| Python     | pep8, isort, ruff  | [.restyled.yaml](https://github.com/project-chip/connectedhomeip/blob/master/.restyled.yaml) (command line),[isort](https://github.com/project-chip/connectedhomeip/blob/master/.isort.cfg) [ruff](https://github.com/project-chip/connectedhomeip/blob/master/ruff.toml) |
```

Comment:
nit: Maybe for readability's sake you can use link "links", i.e.:

```md
Some nice web page [here][link]
 
[link]: https://github.com/project-chip/connectedhomeip/pull/36275/files
```

Also there is no space after the first comma and there is no comma on the second space :D

Reply:
sounds good - updated.